tests/unit/mocks/mock-any-test.js

```diff
@@ -200,4 +200,27 @@ module('MockAny', function(hooks) {
     assert.equal(fooMock.timesCalled, 1, 'fooMock#timesCalled is called once');
     assert.equal(barMock.timesCalled, 1, 'barMock#timesCalled is called once');
   });
+
+  function testWithParamsForMethod(method) {
+    test(`#withParams works with complex parameters for ${method} requests`, async function(assert) {
+      const url     = '/api/post/some-stuff',
+            dataFoo = {foo: 'foo', array: ['a', { b: 123 }, 'c'], obj: { a: 321 } },
```

Comment:
I've changed this test to include objects, arrays, and arrays which include objects.